fabric-keybindings-v0/src/main/java/net/fabricmc/fabric/api/client/keybinding/FabricKeyBinding.java

```diff
@@ -27,23 +31,91 @@
  * {@link KeyBindingRegistry#register(FabricKeyBinding)}!
  */
 public class FabricKeyBinding extends KeyBinding {
+
+    private final Identifier id;
+
 	protected FabricKeyBinding(Identifier id, InputUtil.Type type, int code, String category) {
-		super("key." + id.toString().replace(':', '.'), type, code, category);
+		super(formatKeyName(id), type, code, category);
+		this.id = id;
+	}
+
+	public static String formatKeyName(final Identifier id) {
+	    return String.format("key.%s.%s", id.getNamespace(), id.getPath());
+	}
+
+	/**
+	 * Original identifier used to register this key.
+	 * May be different from the {@link getId()}!
+	 */
+	public Identifier getIdentifier() {
+	    return id;
 	}
 
 	public static class Builder {
-		protected final FabricKeyBinding binding;
 
-		protected Builder(FabricKeyBinding binding) {
-			this.binding = binding;
+		private InputUtil.Type type = InputUtil.Type.KEYSYM;
+
+		private Identifier id = null;
+
+		private int code = GLFW.GLFW_KEY_UNKNOWN;
+
+		private String category = KeyCategory.MISC;
+
+		private Builder() { }
+
+		public Builder id(Identifier keyName) {
+		    this.id = keyName;
+			return this;
+		}
+
+		public Builder category(String category) {
+			this.category = category;
+			return this;
+		}
+
+		public Builder code(int keyCode) {
+			this.code = keyCode;
+			return this;
+		}
+
+		public Builder type(InputUtil.Type type) {
+			this.type = type;
+			return this;
 		}
 
 		public FabricKeyBinding build() {
-			return binding;
+		    Objects.requireNonNull(id, "Keybindings should be created with an identifier.");
+
+		    if (code == GLFW.GLFW_KEY_UNKNOWN) {
+		        throw new IllegalStateException("Keybingings need a default keycode.");
```

Comment:
ok wait a second why am i not allowed to make `unbound` keybinds, this is going to break my mods so hard

Reply:
yeah, you should be able to default to unbound.

Reply:
Okay, remove the throw so you can default to unbound.